webapp/src/main/java/com/box/l10n/mojito/service/machinetranslation/microsoft/MicrosoftMTEngine.java

```diff
@@ -0,0 +1,190 @@
+package com.box.l10n.mojito.service.machinetranslation.microsoft;
+
+import com.box.l10n.mojito.service.machinetranslation.MachineTranslationEngine;
+import com.box.l10n.mojito.service.machinetranslation.TextUnitTranslationGroup;
+import com.box.l10n.mojito.service.machinetranslation.Translation;
+import com.box.l10n.mojito.service.machinetranslation.TranslationSource;
+import com.box.l10n.mojito.service.machinetranslation.TranslationsResponse;
+import com.box.l10n.mojito.service.machinetranslation.microsoft.response.MicrosoftLanguageTranslationDTO;
+import com.box.l10n.mojito.service.machinetranslation.microsoft.response.MicrosoftTextTranslationDTO;
+import io.micrometer.core.annotation.Timed;
+import org.apache.logging.log4j.util.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Component
+@ConditionalOnProperty(value = "l10n.mt.impl", havingValue = "MicrosoftMTEngine")
+public class MicrosoftMTEngine implements MachineTranslationEngine {
+    public static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
+    public static final String OCP_APIM_SUBSCRIPTION_REGION = "Ocp-Apim-Subscription-Region";
+    public static final String X_CLIENT_TRACE_ID = "X-ClientTraceId";
+
+    static Logger logger = LoggerFactory.getLogger(MicrosoftMTEngine.class);
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private MicrosoftMTEngineConfiguration mtEngineConfiguration;
```

Comment:
Auto-wiring the config works and I know the code use that pattern in some place but in that case it might be preferable to keep your service agnostic of spring. 

MicrosoftMTEngine (like S3BlobStorage) seems to be a good candidate to work without spring and be plain Java. 

You'd pass required config via the constructor instead (see S3BlobStorageConfigurationConfiguration). 

Also you end with a central place for setting up the MT engine like (see BlobStorageConfiguration). It is easier to see what the app is doing overall and you get a bit more flexibility (see DatabaseBlobStorageConfiguration it create the blobStorage but also setup job for clean up, etc).

Reply:
Done